### PR TITLE
chore: fix docs linting

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
Docs are no longer being linted, there seems to be some issue with the newer version of eslint. Found this comment saying what version of eslint is compatible with Next 11:
https://github.com/vercel/next.js/issues/30182#issuecomment-965686256

#### Description of how you validated changes

**Before:** 
No linting errors showing, just an error:
```bash
 error - ESLint must be installed in order to run during builds: yarn add --dev eslint
```
See [Build Docs Package step](https://github.com/aws-amplify/amplify-ui/actions/runs/5488767177/jobs/10006266188)

**After:** 
Now seeing our favorite linting errors... 😆 
```bash
./src/pages/[platform]/components/passwordfield/examples/refs.tsx
21:33  Warning: The ref value 'showPasswordButtonRef.current' will likely have changed by the time this effect cleanup function runs. If this ref points to a node rendered by React, copy 'showPasswordButtonRef.current' to a variable inside the effect, and use that variable in the cleanup function.  react-hooks/exhaustive-deps

./src/pages/[platform]/index.page.tsx
49:6  Warning: React Hook React.useEffect has a missing dependency: 'handleScroll'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./src/components/FlutterAuthenticatorExample.tsx
120:6  Warning: React Hook useCallback has a missing dependency: 'id'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./src/components/Fragment.tsx
87:6  Warning: React Hook React.useMemo has a missing dependency: 'platforms'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./src/components/Layout/PageTabLayout.tsx
20:9  Warning: The 'getIndex' function makes the dependencies of useEffect Hook (at line 41) change on every render. To fix this, wrap the definition of 'getIndex' in its own useCallback() Hook.  react-hooks/exhaustive-deps
```

After fixing the errors:
```
✔ No ESLint warnings or errors
```

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by zchenwei — DISMISSED** [review]
Looks like we are missing a few dependencies in hook usage, but they are surprisingly working well 😉

**Review by thaddmt — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

File: docs/src/pages/[platform]/index.page.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ const HomePage = ({ colorMode }) => {
     return () => {
       document.removeEventListener('scroll', handleScroll);
     };
-  }, []);
+  }, [handleScroll]);
```
Comment (calebpollman) [inline docs/src/pages/[platform]/index.page.tsx:49]: Is it intentional to run this effect for every render of `HomePage`?

Comment (reesscot) [reply docs/src/pages/[platform]/index.page.tsx]: fixed

**Review by calebpollman — APPROVED** [review]
🚢

**Review by thaddmt — APPROVED** [review]

**Pull request merged**